src/Bicep.Core/Semantics/DeclarationVisitor.cs

```diff
@@ -161,33 +163,54 @@ public override void VisitProviderDeclarationSyntax(ProviderDeclarationSyntax sy
         {
             base.VisitProviderDeclarationSyntax(syntax);
 
-            TypeSymbol declaredType;
-            if (!features.ExtensibilityEnabled)
+            TypeSymbol resolveProviderSymbol()
```

Comment:
Since you're only using this function once, is there any benefit in this encapsulation?